### PR TITLE
Add Metrics/MethodLength cop for define_method

### PR DESCRIPTION
**Pull request by jekuta**
Metrics/MethodLength cop for `define_method`as described in #4629 

Feature
-----------------
This PR adds a `on_block` method to the cop. It checks whether we are sending the `define method` with a block and if so then passes the node to check_code_lenght method, as we would do with the normal method as well.

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/metrics/method_length_spec.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   subject(:cop) { described_class.new(config) }
   let(:cop_config) { { 'Max' => 5, 'CountComments' => false } }
 
-  it 'rejects a method with more than 5 lines' do
+  it 'rejects a method with more than 5 lines, syntax #1' do
```
Comment (bbatsov) [inline spec/rubocop/cop/metrics/method_length_spec.rb]: I'd surround this with `context`. `syntax #1` and `syntax #2` is too vague.

Comment (jekuta) [reply spec/rubocop/cop/metrics/method_length_spec.rb]: @bbatsov There was some duplication in the spec, so I took the liberty to refactor it a little bit. If you think it's too much, I can revert to the original version, and just add a context around the new example.

**Comment by bbatsov** [comment]
👍

**Comment by AlexWayfer** [comment]
Thank you, guys! :tada:

**Pull request merged**